analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -255,6 +255,9 @@ static SyntaxNode PushPathPositionForTuple(Stack<PathPosition> pathPositions, Tu
                     : null;
         }
 
+        public static bool IsTopLevelMain(this SyntaxNode node) =>
+            node is CompilationUnitSyntax compilationUnit && compilationUnit.Members.Any(x => x.IsKind(SyntaxKind.GlobalStatement));
```

Comment:
This should be on `CompilationUnitSyntax`. Otherwise, it is misleading with the "SyntaxNode".
```suggestion
        public static bool IsTopLevelMain(this CompilationUnitSyntax compilationUnit) =>
            compilationUnit.Members.Any(x => x.IsKind(SyntaxKind.GlobalStatement));
```

Reply:
I thought about this but I wanted to avoid all the casts and additional checks. I see your point though.